e2e/testsuite/testsuite.go

```diff
@@ -611,6 +628,43 @@ func (s *E2ETestSuite) QueryGranterGrants(ctx context.Context, chain *cosmos.Cos
 	return grants.Grants, nil
 }
 
+// IsRelayer returns true if the provided wallet is used by a relayer.
+func (s *E2ETestSuite) IsRelayer(user ibc.Wallet) bool {
+	if s.relayers == nil {
+		return false
+	}
+	return s.relayers[user]
+}
+
+// retryNtimes retries the provided function up to the provided number of attempts.
+func (s *E2ETestSuite) retryNtimes(f func() (sdk.TxResponse, error), attempts int) (sdk.TxResponse, error) {
+	// Ignore account sequence mismatch errors.
+	allowedMessages := []string{"account sequence mismatch"}
+	var err error
+	var resp sdk.TxResponse
+	for i := 0; i < attempts; i++ {
+		// If the response's raw log doesn't contain any of the allowed prefixes we return, else, we retry.
+		resp, err = f()
+		if !containsMessage(resp.RawLog, allowedMessages) {
+			return resp, err
+		}
+		s.T().Logf("retrying tx due to ignored raw log: %s", resp.RawLog)
+	}
+	return resp, err
+}
+
+// containsPrefix returns true if the string s contains any of the prefixes in the slice.
+func containsMessage(s string, messages []string) bool {
+	var hasPrefix bool
+	for _, message := range messages {
+		if strings.Contains(s, message) {
+			hasPrefix = true
```

Comment:
[nit] we can just return here instead of using a boolean and breaking

Reply:
nice catch, we don't actually need a separate variable to keep track of it.